Makefile

```diff
@@ -1,5 +1,5 @@
 test:
-	go test -v --race ./...
+	go test -v --race ./... -timeout=15s
```

Comment:
What is the purpose of this timeout?

Reply:
This change is irrelevant to the PR but it's to prevent our tests to possibly hanging